### PR TITLE
FR: JWK Migration

### PR DESCRIPTION
**Pull request by denopink**
Implements #183 

What's included:
- JWT Migration [183](https://github.com/xmidt-org/scytale/issues/183)
  - updated to use clortho `Resolver` & `Refresher`
  - updated to use clortho `metrics` & `logging`
- Update Config
  - Use [uber/zap](https://github.com/uber-go/zap) for clortho logging
  - Use [xmidt-org/sallust](https://github.com/xmidt-org/sallust) for the zap config unmarshalling 
  - Update auth config for clortho
  
New future work introduced:
- Update to fully use [uber/zap](https://github.com/uber-go/zap) &  [xmidt-org/sallust](https://github.com/xmidt-org/sallust) for logging #186

### CODE REVIEW

**Comment by schmidtw** [comment]
Does this have anything to do with QOS Acks?

**Comment by denopink** [comment]
> Does this have anything to do with QOS Acks?

@schmidtw Nope, here are the PR's related to QOS Acks atm:
- https://github.com/xmidt-org/talaria/pull/236
  - Will merge once qa is done https://github.com/comcast-cl/xmidt/issues/362
- https://github.com/xmidt-org/wrp-go/pull/93
- https://github.com/xmidt-org/wrp-go/pull/81

File: primaryHandler.go [hunk 1]
```diff
@@ -103,30 +113,93 @@ func authChain(v *viper.Viper, logger log.Logger, registry xmetrics.Registry) (a
...
-		if err != nil {
-			return alice.Chain{}, emperror.With(err, "failed to create resolver")
-		}
+	// Instantiate a keyring for refresher and resolver to share
+	kr := clortho.NewKeyRing()
+
+	// Instantiate a fetcher for refresher and resolver to share
+	f, err := clortho.NewFetcher()
+	if err != nil {
+		return alice.Chain{}, emperror.With(err, "failed to create clorth fetcher")
```
Comment (renaz6) [inline primaryHandler.go:126]: Small typo: "clorth" should be "Clortho" here

**Review by renaz6 — COMMENTED** [review]
Looks great! Just caught a couple of typos

File: primaryHandler.go [hunk 2]
```diff
@@ -103,30 +113,93 @@ func authChain(v *viper.Viper, logger log.Logger, registry xmetrics.Registry) (a
...
+	}
+
+	ref, err := clortho.NewRefresher(
+		clortho.WithConfig(jwtVal.Config),
+		clortho.WithFetcher(f),
+	)
+	if err != nil {
+		return alice.Chain{}, emperror.With(err, "failed to create clorth refresher")
```
[the remark this answers is not in the document]
Comment (denopink) [reply primaryHandler.go:134]: copy and paste 😱

**Review by renaz6 — APPROVED** [review]

**Pull request merged**